src/MCPServer/lib/RPCServer.py

```diff
@@ -54,206 +65,374 @@ class NotFoundError(RPCServerError):
+        except KeyError:
+            raise not_found
+        try:
+            choicesAvailableForUnits[job.pk].proceedWithChoice(
+                chain.id, user_id)
+        except IndexError:
+            raise not_found
+
+    def _get_processing_config_fields_handler(self, worker, job):
+        """List processing configuration fields.
+
+        [config]
+        name = getProcessingConfigFields
+        raise_exc = no
+        """
+        return get_processing_fields(self.workflow)
+
+    def _units_statuses_handler(self, worker, job, payload):
+        """Returns the status of units that are of type SIP or Transfer.
+
+        It returns a JSON-encoded objects. Its ``objects`` attribute is an
+        array of objects, each of which represents a single unit. Each unit
+        has a ``jobs`` attribute shoe value is an array of objects, each of
+        which represents a job of the unit.
+
+        [config]
+        name = getUnitsStatuses
+        raise_exc = no
+        """
+        unit_types = {
+            "SIP": (SIP, "unitSIP"),
+            "Transfer": (Transfer, "unitTransfer"),
+        }
+        try:
+            model_attrs = unit_types[payload["type"]]
+            lang = payload["lang"]
+        except KeyError as err:
+            raise UnexpectedPayloadError("Missing parameter: {}".format(err))
+        model = model_attrs[0]
+        sql = """
+        SELECT SIPUUID,
+               MAX(UNIX_TIMESTAMP(createdTime) + createdTimeDec) AS timestamp
+            FROM Jobs
+            WHERE unitType=%s AND NOT SIPUUID LIKE '%%None%%'
+            GROUP BY SIPUUID;"""
+        with connection.cursor() as cursor:
+            cursor.execute(sql, (model_attrs[1],))
+            sipuuids_and_timestamps = cursor.fetchall()
+        # This is a shared data structure, read safely.
+        with choicesAvailableForUnitsLock:
+            jobs_awaiting_for_approval = choicesAvailableForUnits.copy()
+        objects = []
+        for unit_id, timestamp in sipuuids_and_timestamps:
+            item = {
+                # TODO: consolidate this!
+                "id": unit_id, "uuid": unit_id,
+                "timestamp": float(timestamp),
+                "jobs": [],
+            }
+            # Check if hidden (TODO: this method is slow)
+            if model.objects.is_hidden(unit_id):
+                continue
+            jobs = Job.objects.filter(sipuuid=unit_id).order_by("-createdtime")
+            if jobs:
+                item["directory"] = jobs[0].get_directory_name()
+            # Embed accession ID in status data (for DRMC customization)
+            if isinstance(model, Transfer):
+                try:
+                    trfs = Transfer.objects.filter(
+                        file__sip_id=item["uuid"]).distinct()
+                    if trfs:
+                        item["access_system_id"] = trfs[0].access_system_id
+                except Exception:
+                    pass
```

Comment:
Again, I think these are just transplanted lines? We can fix them in future, but we can be more specific with exceptions. It looks like we should just be capturing an ORM exception and a `ValueError`.

Reply:
I'm having a hard time trying to find the set of ORM exceptions that I should be checking, do you have an idea?

Reply:
I feel like it would be `models.DoesNotExist` but I don't know how likely it is, or if it's the only one. Looking at the module, then I see a few more examples, so I think we should leave this for now. `except Exception` - across the code base we have `99 matches across 63 files` so it's not unique.